packages/core/admin/ee/server/services/review-workflows/workflows/content-types.js

```diff
@@ -1,56 +1,76 @@
-  /**
-   * Migrates entities stages. Used when a content type is assigned to a workflow.
-   * @param {*} options
-   * @param {Array<string>} options.srcContentTypes - The content types assigned to the previous workflow
-   * @param {Array<string>} options.destContentTypes - The content types assigned to the new workflow
-   * @param {Workflow.Stage} options.stageId - The new stage to assign the entities to
-   */
-  async migrate({ srcContentTypes = [], destContentTypes, stageId }) {
-    const { created, deleted } = diffContentTypes(srcContentTypes, destContentTypes);
-
-    await mapAsync(created, async (uid) => {
-      // If it was assigned to another workflow, transfer it from the previous workflow
-      const srcWorkflow = await getService('workflows').getAssignedWorkflow(uid);
-      if (srcWorkflow) {
-        // Updates all existing entities stages links to the new stage
-        await getService('stages').updateEntitiesStage(uid, { toStageId: stageId });
-        return this.transferContentType(srcWorkflow, uid);
-      }
-
-      // Create new stages links to the new stage
-      return getService('stages').updateEntitiesStage(uid, {
-        fromStageId: null,
-        toStageId: stageId,
+module.exports = ({ strapi }) => {
+  const contentManagerContentTypeService = strapi
+    .plugin('content-manager')
+    .service('content-types');
+  const stagesService = getService('stages', { strapi });
+  const workflowsService = getService('workflows', { strapi });
+
+  return {
+    /**
+     * Migrates entities stages. Used when a content type is assigned to a workflow.
+     * @param {*} options
+     * @param {Array<string>} options.srcContentTypes - The content types assigned to the previous workflow
+     * @param {Array<string>} options.destContentTypes - The content types assigned to the new workflow
+     * @param {Workflow.Stage} options.stageId - The new stage to assign the entities to
+     */
+    async migrate({ srcContentTypes = [], destContentTypes, stageId }) {
+      const { created, deleted } = diffContentTypes(srcContentTypes, destContentTypes);
+
+      await mapAsync(created, async (uid) => {
+        // If it was assigned to another workflow, transfer it from the previous workflow
+        const srcWorkflow = await workflowsService.getAssignedWorkflow(uid);
+        if (srcWorkflow) {
+          // Updates all existing entities stages links to the new stage
+          await stagesService.updateEntitiesStage(uid, { toStageId: stageId });
+          return this.transferContentType(srcWorkflow, uid);
+        }
+        // Merge options in the configuration as the configuration service use a destructuration merge which doesn't include nested objects
+        const modelConfig = await contentManagerContentTypeService.getConfiguration(uid);
+        await contentManagerContentTypeService.updateConfiguration(
+          { uid },
+          { options: merge(modelConfig.options, { reviewWorkflows: true }) }
+        );
```

Comment:
I think it could be a good idea to encapsulate this in a method,

```suggestion
        await updateContentTypeConfig({reviewWorkflows: true})
```

It makes the intent of this block a little bit clearer, and you don't need to understand you are using the content manager content type service. wdyt?